src/nix/fmt.md

```diff
@@ -0,0 +1,30 @@
+R""(
+
+# Examples
+
+- Format one or more files or directories:
+
+  With [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt):
+
+  ```console
+  # nix fmt --formatter github:nixpkgs#nixpkgs-fmt ./folder ./file.nix
```

Comment:
Is there a mechanism to select a default formatter for a project?

I don't really get how the current examples are different than `nix run github:nixpkgs#nixpkgs-fmt ./folder ./file.nix` for example.

Reply:
as with any `nix.conf` value you should be able to set it via `nixConfig`. At least that is how I understand it. @kamadorueda correct?

```nix
# flake.nix
{
  description = "example";
  nixConfig.formatter = "github:foo/bar/commit?hash=123";
}
```

Reply:
`?hash` is problematic as it suggests that we'll have to reinvent `inputs` for it.

It being outside `outputs` also means that it can't be abstracted over. I'd like to manage tooling configuration in a central flake and reference that in other projects.

Using `outputs.nixFormatter.${system}` will allow that.

It _must not_ be a `nix.conf` option, because that will inevitably lead to formatting churn as contributors with different versions reformat code in projects that do not pin the formatter. For the same reason, it must not be hardcoded in `nix`, because not everyone runs the same Nix version.